winpr/libwinpr/utils/ntlm.c

```diff
@@ -110,7 +113,11 @@ BYTE* NTOWFv2W(LPWSTR Password, UINT32 PasswordLength, LPWSTR User,
 
 	/* Compute the HMAC-MD5 hash of the above value using the NTLMv1 hash as the key, the result is the NTLMv2 hash */
 	if (!winpr_HMAC(WINPR_MD_MD5, NtHashV1, 16, buffer, UserLength + DomainLength, NtHash, WINPR_MD4_DIGEST_LENGTH))
+	{
+		if (allocate)
+			free(NtHash);
 		result = NULL;
+	}
```

Comment:
If `NtHash` was allocated here we have to free it in any case at the end, correct?

Reply:
This is more or less the same as `freerdp_bitmap_planar_compress_plane_rle`. `NtHash` may be passed in, or `NULL` is passed in. But it seems that there is a bug, because the newly allocated `NtHash` should be also returned...

Reply:
It seems that the same problems have also other functions in that file. May I change all the functions in that file to return `BOOL` to be obvious that you have to provide non-`NULL`?